hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddCacheConfigMessageTask.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.dynamicconfig;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.DynamicConfigAddCacheConfigCodec;
+import com.hazelcast.config.CachePartitionLostListenerConfig;
+import com.hazelcast.config.CacheSimpleConfig;
+import com.hazelcast.config.CacheSimpleConfig.ExpiryPolicyFactoryConfig;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.instance.Node;
+import com.hazelcast.internal.dynamicconfig.AddDynamicConfigOperationFactory;
+import com.hazelcast.nio.Connection;
+import com.hazelcast.spi.OperationFactory;
+
+import java.util.List;
+
+public class AddCacheConfigMessageTask
+        extends AbstractAddConfigMessageTask<DynamicConfigAddCacheConfigCodec.RequestParameters> {
+
+    public AddCacheConfigMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected DynamicConfigAddCacheConfigCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return DynamicConfigAddCacheConfigCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        return DynamicConfigAddCacheConfigCodec.encodeResponse();
+    }
+
+    @Override
+    protected OperationFactory getOperationFactory() {
+        CacheSimpleConfig config = new CacheSimpleConfig();
+        config.setAsyncBackupCount(parameters.asyncBackupCount);
```

Comment:
Then maybe it's a good idea to create a JavaDoc and link all holder and task etc. classes which are connected to that config class. So you directly know that a config class is part of the dynamic config feature and were updates have to be done.

```java
/**
 * Simple configuration to hold parsed XML configuration.
 * <p>
 * CacheConfig depends on the JCache API. If the JCache API is not in the classpath,
 * you can use CacheSimpleConfig as a communicator between the code and CacheConfig.
 * <p>
 * This class is used for the dynamic creation of configurations during runtime.
 * Please check the linked classes when modifying this class.
 *
 * @see AddCacheConfigMessageTask
 */
```
It would also be good to have a short note in the JavaDoc, which class is allowed to be changed (e.g. what to do if I add a new field).